### ISSUE TITLE
请问转存出来的文件可以用C++调用吗？

### ISSUE THREAD

**Issue by HaoDreamlong** [grimoire/mmdetection-to-tensorrt#23]
请问这里转出来的文件如何用tensorRT的C++接口来调用并加载模型，可行吗and how？

**Comment by grimoire** [comment]
Yes, please read [use in c++](https://github.com/grimoire/mmdetection-to-tensorrt/blob/master/docs/getting_started.md#use-in-c) for detail.

**Comment by HaoDreamlong** [comment]
thanks for your answer, but I cant find the "libamirstan_plugin.so"   in this project or "Amirstan_plugin" project, can you tell me where I can get this?

**Comment by grimoire** [comment]
It is in `${AMIRSTAN_PLUGIN_PATH}/build/lib` if you have build the plugin follow the readme.

**Comment by HaoDreamlong** [comment]
十分感谢，弄好了，请问我想导入python存的engine的话，是否有api可以直接调用？还是要用c++读取文件？麻烦您了。

**Comment by grimoire** [comment]
Use [deepstream support](https://github.com/grimoire/mmdetection-to-tensorrt/blob/master/docs/getting_started.md#deepstream-support) if you want to do stream inference. Or use c++ api like [official one](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#perform_inference_c). 
Preprocess and postprocess are not in the engine. Should be implemented by your self.

**Comment by ShaneYS** [comment]
> Use [deepstream support](https://github.com/grimoire/mmdetection-to-tensorrt/blob/master/docs/getting_started.md#deepstream-support) if you want to do stream inference. Or use c++ api like [official one](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#perform_inference_c).
> Preprocess and postprocess are not in the engine. Should be implemented by your self.

In SSD, does postprocess refer to operations like decode and nms?
How should we understand postprocess for model deployment?  Operations beside conv, pool, fc?

**Comment by grimoire** [comment]
@ShaneYS  The only **postprocess** you need to do is 
```
bboxes = bboxes/scale_factor
```

**Comment by ShaneYS** [comment]
> 
> 
> @ShaneYS The only **postprocess** you need to do is
> 
> ```
> bboxes = bboxes/scale_factor
> ```

谢谢
还是有点问题。
在导出onnx或者TensorRT时，什么类型的操作会被导出，什么类型的操作不会被导出？
为什么在很多pytorch转TensorRT的代码中，都说nms操作无法导出，需要自己实现？

**Comment by grimoire** [comment]
@ShaneYS 
[Here](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Layers.html) is the TensorRT supported layers.
Ops like nms does not supported can be implemented as [TensorRT Plugin](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#extending). That's why this project need `amirstan_plugin`.

**Comment by ShaneYS** [comment]
> @ShaneYS
> [Here](https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Graph/Layers.html) is the TensorRT supported layers.
> Ops like nms does not supported can be implemented as [TensorRT Plugin](https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#extending). That's why this project need `amirstan_plugin`.

Got it, thanks!

**Comment by 18871271658** [comment]
> It is in `${AMIRSTAN_PLUGIN_PATH}/build/lib` if you have build the plugin follow the readme.

if i cant find the libamirstan_plugin.so in `${AMIRSTAN_PLUGIN_PATH}/build/lib` after cmake-DTENSORRT_DIR=${TENSORRT_DIR} .. and make -j10, what should i do? please.

**Comment by 18871271658** [comment]
> 十分感谢，弄好了，请问我想导入python存的engine的话，是否有api可以直接调用？还是要用c++读取文件？麻烦您了。

请问我cmake 和make后在build/lib找不到那个libamirstan_plugin.so文件怎么办，你可以分享一下那个文件吗
